projects/gnomad/src/client/GenePage/GnomadConstraintTable.js

```diff
@@ -98,7 +98,17 @@ Graph.propTypes = {
   upper: PropTypes.number.isRequired,
 }
 
-const renderNumber = (num, precision) => (num === null ? '—' : num.toFixed(precision))
+const renderRoundedNumber = (num, precision) => {
+  if (num === null) {
```

Comment:
Would it make sense to return '-' by default, and only return tooltipanchor if it is what we expect (num/float)? i.e.
psuedocode:
```
if typeof num is number:
  return tooltip
return '-'
```
Kind of like whitelisting instead of blacklisting.

Reply:
I would argue that if something of an unexpected type gets passed to this function, then it's a sign of some logic mistake and `num.toFixed(precision)` throwing an error and calling attention to the problem is preferable to accepting it and returning "-".

The values which are passed to this function are properties of gnomAD gene constraint, which GraphQL coerces to numbers (or null) when they are returned from the API.
https://github.com/macarthur-lab/gnomadjs/blob/c239ac4deaaa9846359ea889e18932634d4f1416/packages/api/src/schema/datasets/gnomad_r2_1/GnomadConstraintType.js#L3-L34

Reply:
Agreed that identifying invalid forms and bringing this up to devs is best (i.e. fail silently and record error). If this throws and exception, is it silently caught and reported?

Reply:
If it throws, it's not caught, it prevents the component from rendering. But that same risk is present in every component. PropTypes will warn about an incorrect type in development, but won't prevent it from causing an error. Verifying the type of every argument/prop at run-time would be a huge amount of work.

With the exception of variants being created as Immutable.js Records with a defined set of fields, the API response schema is not validated on the client side.

Reply:
Makes more sense after talking in person... Validating the API response is a good place to start for greater type safety since the API can be deployed separately from the browser UI. Also filed #396 for tracking exceptions in the browser.